packages/jest-serializer/src/index.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import fs from 'fs';
+import v8 from 'v8';
+
+import type {Path} from 'types/Config';
+
+type IOCallback = (?Error, ?any) => void;
+
+// JSON and V8 serializers are both stable when it comes to compatibility. The
+// current JSON specification is well defined in RFC 8259, and V8 ensures that
+// the versions are compatible by encoding the serialization version in the own
+// generated buffer.
+
+// In memory functions.
+
+export function serialize(content: any): Buffer {
+  return v8.serialize
+    ? v8.serialize(content)
+    : Buffer.from(JSON.stringify(content));
+}
+
+export function deserialize(buffer: Buffer): any {
+  return v8.deserialize
+    ? v8.deserialize(buffer)
+    : Buffer.from(JSON.stringify(buffer.toString('utf8')));
+}
+
+// Synchronous filesystem functions.
+
+export function readFileSync(file: Path): any {
+  return v8.deserialize
+    ? v8.deserialize(fs.readFileSync(file))
+    : JSON.parse(fs.readFileSync(file, 'utf8'));
+}
+
+export function writeFileSync(file: Path, content: any) {
+  return v8.serialize
+    ? fs.writeFileSync(file, v8.serialize(content))
+    : fs.writeFileSync(file, JSON.stringify(content), 'utf8');
+}
+
+// Asynchronous filesystem functions.
+
+export function readFile(file: Path, callback: IOCallback) {
```

Comment:
`_if_` we expose an async API, I'd prefer it to be promise instead of callback based

Reply:
I just killed the async one 😜. But you are right; it should have been `Promise` based. I just followed the standard Node JS format.

The reason for killing it is because we don't really need it anywhere: neither in `jest-haste-map`, nor in `jest-worker` (coming soon), nor in Metro, where we will use this module as well.